packages/react-server/src/ReactFizzServer.js

```diff
@@ -4095,6 +4155,22 @@ export function startFlowing(request: Request, destination: Destination): void {
     return;
   }
   request.destination = destination;
+
+  // In some cases it is possible to startFlowing before the first performWork has finished.
+  // We need to emitEarlyPreloads before we start flowing in case the host can only do this
+  // once and it needs to be done before we start streaming. If this happens it is almost
+  // certain that there will actually be nothing to preload, regardless we do this here for
+  // correctness
+  const shellComplete =
+    request.trackedPostpones === null
+      ? // Render Request, we define shell complete by the pending root tasks
+        request.pendingRootTasks === 0
+      : // Prerender Request, we define shell complete by completedRootSegemtn
+      request.completedRootSegment === null
+      ? request.pendingRootTasks === 0
+      : request.completedRootSegment.status !== POSTPONED;
+  emitEarlyPreloads(request.renderState, request.resumableState, shellComplete);
```

Comment:
This can potentially happen a lot when you drain a queue often. Unfortunate to add all these extra checks here.

Additionally, if this is happening in a drain event, you might be in a bad context to do something about it.

This is also impossible in all APIs but Node.

I wonder if we should just move this into its own function for Node or something.

Reply:
yeah makes sense